cli/cmd/add_resource.go

```diff
@@ -83,27 +85,47 @@ keptn add-resource --project=keptn --service=keptn-control-plane --all-stages --
 			return internal.OnAPIError(err)
 		}
 
-		// Handle different cases of adding resource to a projects default branch, stage branch, and/or service sub-directory
-		if isStringFlagSet(addResourceCmdParams.Service) && isBoolFlagSet(addResourceCmdParams.AllStages) {
-			// add to all stages
-			logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" to all stages of project "+*addResourceCmdParams.Project, logging.InfoLevel)
-		} else if areStringFlagsSet(addResourceCmdParams.Service, addResourceCmdParams.Stage) {
-			// add to service and stage
-			logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" to service "+*addResourceCmdParams.Service+" in stage "+*addResourceCmdParams.Stage+" in project "+*addResourceCmdParams.Project, logging.InfoLevel)
-		} else if !isStringFlagSet(addResourceCmdParams.Service) && isStringFlagSet(addResourceCmdParams.Stage) {
-			// service is empty, add to stage
-			logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" to stage "+*addResourceCmdParams.Stage+" in project "+*addResourceCmdParams.Project, logging.InfoLevel)
-		} else if !isStringFlagSet(addResourceCmdParams.Service) && !isStringFlagSet(addResourceCmdParams.Stage) {
-			// service and stage are empty, add to default branch
-			logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" to project "+*addResourceCmdParams.Project, logging.InfoLevel)
-		} else {
-			return errors.New("Flag 'stage' is missing")
-		}
-
 		if !mocking {
-			if addResourceCmdParams.AllStages != nil && *addResourceCmdParams.AllStages {
-				// Upload to all stages
-				// get stages
+			if !isStringFlagSet(addResourceCmdParams.Service) && !isBoolFlagSet(addResourceCmdParams.AllStages) && !isStringFlagSet(addResourceCmdParams.Stage) {
+				// project resource
+				logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" as a project resource to project "+*addResourceCmdParams.Project, logging.InfoLevel)
+				_, errorObj := api.ResourcesV1().CreateResources(*addResourceCmdParams.Project, "", "", resources)
+				if errorObj != nil {
+					return errors.New("Resource " + *addResourceCmdParams.Resource + " could not be uploaded as a project resource: " + *errorObj.Message)
+				}
+			} else if !isStringFlagSet(addResourceCmdParams.Service) && isBoolFlagSet(addResourceCmdParams.AllStages) {
+				// stage resource to all stages
+				logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" as a stage resource to all stages for project "+*addResourceCmdParams.Project, logging.InfoLevel)
+				api, err := internal.APIProvider(endPoint.String(), apiToken)
```

Comment:
i belive the initialization of `api, err` can be removed since that already happened some lines above

Reply:
fixed